frontends/p4/typeChecking/typeUnification.cpp

```diff
@@ -53,7 +53,7 @@ bool TypeUnification::unifyFunctions(const IR::Node* errorPosition,
         }
     }
 
-    if (dest->parameters->size() != src->arguments->size()) {
+    if (dest->parameters->size() < src->arguments->size()) {
```

Comment:
There are many other places where the argument list and the parameter list are matched; the most important one is in MethodInstance::MethodCallDescription, but unfortunately this class was created late and there's still of other code which manually iterates over these lists. Adding optional parameters will require revisiting all of these places. I'd really rather have them bound with _ values.

Moreover, in the absence of any test inputs it won't be easy to maintain this code.